stringutil/stringutil.go

```diff
@@ -237,4 +237,40 @@ func NumericOnly(s string) string {
 		}
 	}
 	return string(nRunes)
-}
\ No newline at end of file
+}
+
+// ObfuscateString with asterisks in place of all characters except the first 3
+// characters, fewer if string is shorter than 5 characters
+func ObfuscateString(raw string) string {
```

Comment:
This should probably be 'ObfuscateRight' or take a direction parameter since some things (e.g. credit card numbers XXXX XXXX XXXX 1234) need to be obfuscated from the left.

Reply:
the automatic is nice, but generally you want to provide the ability for the end user to be explicit. For instance allow for specifying the obfuscation character, direction, and number of charcters which would then try and obfuscate as many characters as possible padding the rest. So, 
obfuscate("foo bar", 2, left, '*') = "**o bar"
obfuscate("foo", 5, left, 'X') = "XXXXX"

This would get most use cases with any additional use cases easily being constructed from it.

Reply:
Very good point, I should have been thinking about this more in the context of the stringutil package rather than just what I wanted to use it for. I've pushed updates and removed the email and phone specific helpers since those will make more sense if they are housed where they are used in core. Thanks!

Reply:
Also tried out the swanky new table driven test format. Not that it was at all beneficial for such simple tests